akka-stream/src/main/scala/akka/stream/Materializer.scala

```diff
@@ -237,6 +259,73 @@ object Materializer {
   def createMaterializer(systemProvider: ClassicActorSystemProvider): Materializer =
     apply(systemProvider)
 
+  /**
+   * Scala API: Create a new materializer that will stay alive as long as the system does or until it is explicitly stopped.
+   *
+   * The provided [[Attributes]] will be applied (with already-applied attributes taking precedence) to the streams materialized by the created materializer.
+   *
+   * It is generally advised to limit the number of system level materializers created.
+   */
+  def apply(systemProvider: ClassicActorSystemProvider, defaultAttributes: Attributes): Materializer =
+    new MaterializerWithAttributes(apply(systemProvider), defaultAttributes)
+
+  /**
+   * Java API: Create a new materializer that will stay alive as long as the system does or until it is explicitly stopped.
+   *
+   * The provided [[Attributes]] will be applied (with already-applied attributes taking precedence) to the streams materialized by the created materializer.
+   *
+   * It is generally advised to limit the number of system level materializers created.
+   */
+  def createMaterializer(systemProvider: ClassicActorSystemProvider, defaultAttributes: Attributes): Materializer =
+    new MaterializerWithAttributes(createMaterializer(systemProvider), defaultAttributes)
+}
+
+/** The constructor is INTERNAL API */
+final class MaterializerWithAttributes @InternalApi private[stream] (
```

Comment:
I think we should update the existing implementation rather than decorate/delegate this. A new internal `ActorMaterializer.apply` overload that takes attributes and combines those with the defaults from `materializerSettings.toAttributes`.